src/Zicht/Bundle/HtmldevBundle/Service/SvgService.php

```diff
@@ -4,77 +4,86 @@
  */
 namespace Zicht\Bundle\HtmldevBundle\Service;
 
+use Psr\Log\LoggerInterface;
+use Psr\SimpleCache\CacheInterface;
+
 /**
  * Class that reads the contents of an SVG file.
  */
 class SvgService implements SvgServiceInterface
 {
     /** @var string */
-    private $htmldevDirectory;
+    private $baseDir;
+    /** @var LoggerInterface */
+    private $logger;
+    /** @var CacheInterface */
+    private $cache;
 
     /**
-     * Initializes a new instance of the SvgService.
-     *
-     * @param string $htmldevDirectory
+     * @param string $baseDir
+     * @param CacheInterface $cache
+     * @param LoggerInterface $logger
      */
-    public function __construct($htmldevDirectory)
+    public function __construct(string $baseDir, CacheInterface $cache, LoggerInterface $logger)
     {
-        $this->htmldevDirectory = $htmldevDirectory;
+        $this->baseDir = $baseDir;
+        $this->cache = $cache;
+        $this->logger = $logger;
     }
 
     /**
      * {@inheritDoc}
      */
     public function getSvg($name, $width, $height, $viewboxX, $viewboxY, $cssClasses, $attributes, $title, $directory)
     {
-        $fileName = sprintf('%s/%s/%s.svg', $this->htmldevDirectory, $directory, $name);
-        if (!is_file($fileName)) {
-            return null;
-        }
-
-        $d = new \DOMDocument();
-        $d->load($fileName);
-
-        $svg = $d->getElementsByTagName('svg')->item(0);
-
-        /**
-         * Set dimensions to ensure proper rendering across different browsers.
-         */
-        if (!empty($width)) {
-            $this->setSvgAttribute($svg, 'width', $this->addUnit($width, 'px'), true);
-        }
-
-        if (!empty($height)) {
-            $this->setSvgAttribute($svg, 'height', $this->addUnit($height, 'px'), true);
-        }
-
-        if (!empty($viewboxX) && !empty($viewboxY)) {
-            $this->setSvgAttribute($svg, 'viewBox', sprintf('0 0 %s %s', $this->removeUnit($viewboxX, 'px'), $this->removeUnit($viewboxY, 'px')), true);
-        }
-
-        $this->setSvgAttribute($svg, 'preserveAspectRatio', 'xMidYMid meet');
-
-        /**
-         * Add CSS classes for extra theming or positioning.
-         */
-        if (is_array($cssClasses)) {
-            $this->setSvgAttribute($svg, 'class',  implode('  ', $cssClasses));
-        }
-
-        if (!empty($title)) {
-            $svg->appendChild($d->createElement('title', $title));
-        }
-
-        /**
-         * Set accessibility attributes.
-         */
-        if (is_array($attributes)) {
-            foreach ($attributes as $attr => $value) {
-                $svg->setAttribute($attr, $value);
+        $key = sha1(json_encode(func_get_args()));
+        if (null === $out = $this->cache->get($key)) {
```

Comment:
Wat ik al in Slack postte: Heb er even doorheen gescrold, op zich een goede toevoeging :+1:. Ik zou er bijv. morgen even beter naar moeten kijken. Ik heb de "backend" van de HTMLDEV bundle nog nooit ingekeken...

Wel wil ik op deze constructie alvast reageren. Ik vond het in eerste instantie onduidelijk wat er nu precies gebeurde in geval van dat hij niet de 'if' in ging. Alleen een log wegschrijven? Na wat heen en weer scrollen begreep ik het.

Ik denk dat het veel duidelijker is als je doet:
```php
if (null !== $out = $this->cache->get($key)) {
    $this->logger->info('svg ' . $name . ' loaded from cache');
    return $out;
}

# rest v.d. code (indent -1)
# ...
# ...
$this->cache->set($key, $out);

return $out;
```